pkg/controller/workload/workload_event.go

```diff
@@ -88,7 +89,7 @@ func newAckRequest(rsp *service_discovery_v3.DeltaDiscoveryResponse) *service_di
 		ResourceNamesSubscribe: []string{},
 		ResponseNonce:          rsp.GetNonce(),
 		ErrorDetail:            nil,
-		Node:                   config.getNode(),
+		Node:                   config.GetConfig().GetNode(),
 	}
 }
 
```

Comment:
rsp.RemovedResources branch which in function processWorkloadResponse， we also need to consider different types ；like： AddressType or authorization ..

if rsp.RemovedResources != nil {
		log.Debugf("RemovedResources %s\n", rsp.RemovedResources)
		err = handleDeleteResponse(rsp)
	}

Reply:
Fixed